spec/validate.go

```diff
@@ -0,0 +1,82 @@
+package spec
+
+import (
+	"github.com/lestrrat/go-jsschema"
+	"github.com/lestrrat/go-jsval"
+	"github.com/lestrrat/go-jsval/builder"
+)
+
+func GetValidatorForOpenAPI3Schema(oaiSchema *Schema) (*jsval.JSVal, error) {
```

Comment:
It would be _super_ great if we could get a basic test for this function (or maybe for `getJSONSchemaForOpenAPI3Schema` where most of the heavy logic lives). Maybe one that manually defines a simplistic spec struct that contains a `Nullable` field and shows how it's converted to an equivalent JSON schema with a `null` type. What do you think?

Reply:
We have a couple of basic tests in `spec_test.go`. Think we need more? Also, should they be in `schema_test.go` instead?

Reply:
> We have a couple of basic tests in spec_test.go. Think we need more?

Hm, actually would you mind just moving the two `TestValidator_*` tests from `spec_test.go` over to a new `validate_test.go`?

> Also, should they be in schema_test.go instead?

I think it's usually good convention to have `*_test.go` files that are named after a `*.go` file of the same name in the package. IMO it would best for now here to just have two separate test files. You can still easily test the whole package in one command:

``` sh
go test ./spec
```

Reply:
D'oh, I meant to write `validate_test.go` (`validate.go` used to be named `schema.go`). Will do.